File: Src/java/cql-to-elm/src/jvmTest/java/org/cqframework/cql/cql2elm/operators/CqlComparisonOperatorsTest.java
package org.cqframework.cql.cql2elm.operators;

import static kotlinx.io.CoreKt.buffered;
import static kotlinx.io.JvmCoreKt.asSource;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;
import org.cqframework.cql.cql2elm.CqlTranslator;
import org.cqframework.cql.cql2elm.LibraryManager;
import org.cqframework.cql.cql2elm.ModelManager;
import org.hl7.elm.r1.ExpressionDef;
import org.hl7.elm.r1.Library;
import org.junit.jupiter.api.BeforeAll;

/**
 * Created by Bryn on 12/30/2016.
 */
class CqlComparisonOperatorsTest {

    // NOTE: The CQL for this test is taken from an engine testing suite that produced a particular issue with the
    // operatorMap. This library will not translate successfully without the proper fix in place.
    // So this test only needs to validate that the library translates successfully.

    private static Map<String, ExpressionDef> defs;

    @BeforeAll
    static void setup() throws IOException {
        ModelManager modelManager = new ModelManager();
        CqlTranslator translator = CqlTranslator.fromSource(
                buffered(asSource(CqlComparisonOperatorsTest.class.getResourceAsStream("../OperatorTests/CqlComparisonOperators.cql"))),
                new LibraryManager(modelManager));
        assertThat(translator.getErrors().size(), is(0));
        Library library = translator.toELM();
        defs = new HashMap<>();
        if (library.getStatements() != null) {
            for (ExpressionDef def : library.getStatements().getDef()) {
                defs.put(def.getName(), def);
            }
        }
    }

    // @Test
    // public void testEqual() {
    //    ExpressionDef def = defs.get("SimpleEqNullNull");
    //    assertThat(def, hasTypeAndResult(Equal.class, "System.Boolean"));
    // }
}

File: cql-to-elm/src/jvmTest/kotlin/org/cqframework/cql/cql2elm/operators/CqlComparisonOperatorsTest.kt
package org.cqframework.cql.cql2elm.operators

import java.io.IOException
import kotlinx.io.asSource
import kotlinx.io.buffered
import org.cqframework.cql.cql2elm.CqlTranslator.Companion.fromSource
import org.cqframework.cql.cql2elm.LibraryManager
import org.cqframework.cql.cql2elm.ModelManager
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.hl7.elm.r1.ExpressionDef
import org.junit.jupiter.api.BeforeAll

/** Created by Bryn on 12/30/2016. */
internal object CqlComparisonOperatorsTest {
    // NOTE: The CQL for this test is taken from an engine testing suite that produced a particular
    // issue with the
    // operatorMap. This library will not translate successfully without the proper fix in place.
    // So this test only needs to validate that the library translates successfully.
    private var defs: MutableMap<String?, ExpressionDef?>? = null

    @JvmStatic
    @BeforeAll
    @Throws(IOException::class)
    fun setup() {
        val modelManager = ModelManager()
        val translator =
            fromSource(
                CqlComparisonOperatorsTest::class
                    .java
                    .getResourceAsStream("../OperatorTests/CqlComparisonOperators.cql")!!
                    .asSource()
                    .buffered(),
                LibraryManager(modelManager),
            )
        assertThat(translator.errors.size, `is`(0))
        val library = translator.toELM()
        defs = HashMap()
        if (library!!.statements != null) {
            for (def in library.statements!!.def) {
                defs!![def.name] = def
            }
        }
    } // @Test
    // public void testEqual() {
    //    ExpressionDef def = defs.get("SimpleEqNullNull");
    //    assertThat(def, hasTypeAndResult(Equal.class, "System.Boolean"));
    // }
}
